packages/datadog-plugin-rhea/src/index.js

```diff
@@ -0,0 +1,210 @@
+'use strict'
+
+const dd = Symbol('datadog')
+const circularBufferConstructor = Symbol('circularBufferConstructor')
+
+function createWrapSend (tracer, config, instrumenter) {
+  return function wrapSend (send) {
+    return function sendWithTrace (msg, tag, format) {
+      patchCircularBuffer(this, instrumenter)
+      const name = this.options && this.options.target && this.options.target.address
+        ? this.options.target.address : 'amq.topic'
+      const { host, port } = getHostAndPort(this.connection)
+      return tracer.trace('amqp.send', {
+        tags: {
+          'component': 'rhea',
+          'resource.name': name,
+          'service.name': config.service || `${tracer._service}-amqp-producer`,
+          'span.kind': 'producer',
+          'amqp.link.target.address': name,
+          'amqp.link.role': 'sender',
+          'out.host': host,
+          'out.port': port
+        }
+      }, (span, done) => {
+        addDeliveryAnnotations(msg, tracer, span)
+        const delivery = send.apply(this, arguments)
+        delivery[dd] = { done, span }
+        return delivery
+      })
+    }
+  }
+}
+
+function createWrapConnectionDispatch (tracer, config) {
+  return function wrapDispatch (dispatch) {
+    return function dispatchWithTrace (eventName, obj) {
+      if (eventName === 'disconnected' && this.local_channel_map) {
```

Comment:
What happens if `this.local_channel_map`? Right now it looks like the spans will stay in memory. Is there any way this could be handled to make sure it cannot happen?